gammapy/catalog/tests/test_xmatch.py

```diff
@@ -10,13 +10,19 @@
     table_xmatch_circle_criterion,
     table_xmatch,
 )
+from ...utils.random import check_random_state
 
 
 def test_catalog_xmatch_circle():
+    # initialise random number generator
     np.random.seed(0)
+    #rng = check_random_state(0)
+
     catalog = make_catalog_random_positions_sphere(size=100, center='Milky Way')
     catalog['Source_Name'] = ['source_{:04d}'.format(_) for _ in range(len(catalog))]
     catalog['Association_Radius'] = Angle(10 * np.random.random(len(catalog)), unit='deg')
+    #catalog['Association_Radius'] = Angle(rng.uniform(0, 10, len(catalog)), unit='deg')
+    # TODO: not working with rng!!! (but why did it work before??!!!)
```

Comment:
This was deterministic before because of the `np.random.seed(0)` call at the top of the `test_catalog_xmatch_circle` function. Because we were using the global numpy rng everywhere, it was deterministic.

As you say, now that we're using the pattern that all functions using rng calls take the random_state parameter, yes, you have to pass it around.